plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -365,6 +365,7 @@ private static Optional<String> getQueryId(io.trino.plugin.hive.metastore.Table
     public void unregisterTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
         dropTableFromMetastore(schemaTableName);
+        invalidateTableCache(schemaTableName);
```

Comment:
Can this and folowing be moved to `dropTableFromMetastore` ?

Reply:
https://github.com/trinodb/trino/pull/19757#discussion_r1395717687